src/charm.py

```diff
@@ -693,40 +696,151 @@ def _activate_charm(self) -> None:
-                working_dir=DISCOURSE_PATH,
-                user=CONTAINER_APP_USERNAME,
-                environment=self._create_discourse_environment_settings(),
-                timeout=60,
+        if not container.can_connect():
+            event.fail("Unable to connect to container, container is not ready")
+            return
+
+        email = event.params["email"]
+
+        if not self._user_exists(email):
+            event.fail(f"User with email {email} does not exist")
+            return
+
+        process = container.exec(
+            [
+                os.path.join(DISCOURSE_PATH, "bin/bundle"),
+                "exec",
+                "rake",
+                "admin:create",
+            ],
+            stdin=f"{email}\nn\nY\n",
+            working_dir=DISCOURSE_PATH,
+            user=CONTAINER_APP_USERNAME,
+            environment=self._create_discourse_environment_settings(),
+            timeout=60,
+        )
+        try:
+            process.wait_output()
+            event.set_results({"user": email})
+        except ExecError as ex:
+            event.fail(
+                f"Failed to make user with email {email} an admin: {ex.stdout}"  # type: ignore
             )
-            try:
-                process.wait_output()
-                event.set_results({"user": f"{email}"})
-            except ExecError as ex:
+
+    def _on_create_user_action(self, event: ActionEvent) -> None:
+        """Create a new user in Discourse.
+
+        Args:
+            event: Event triggering the create_user action.
+        """
+        container = self.unit.get_container(CONTAINER_NAME)
+        if not container.can_connect():
+            event.fail("Unable to connect to container, container is not ready")
+            return
+
+        email = event.params["email"]
+        password = self._generate_password(16)
+
+        if self._user_exists(email):
+            event.fail(f"User with email {email} already exists")
+            return
+
+        # Admin flag is optional, if it is true, the user will be created as an admin
+        admin_flag = "Y" if event.params.get("admin") else "N"
+
+        process = container.exec(
+            [
+                os.path.join(DISCOURSE_PATH, "bin/bundle"),
+                "exec",
+                "rake",
+                "admin:create",
+            ],
+            stdin=f"{email}\n{password}\n{password}\n{admin_flag}\n",
+            working_dir=DISCOURSE_PATH,
+            user=CONTAINER_APP_USERNAME,
+            environment=self._create_discourse_environment_settings(),
+            timeout=60,
+        )
+        try:
+            process.wait_output()
+            event.set_results({"user": email, "password": password})
+        except ExecError as ex:
+            event.fail(f"Failed to make user with email {email}: {ex.stdout}")  # type: ignore
+            return
+
+        if event.params.get("admin") or not event.params.get("active"):
```

Comment:
I'm not sure that I get that condition. Why do we skip activation if admin is True ?
Can you add a comment for that ?